### PR TITLE
feat(engine): Change matrix to numpy

### PR DESCRIPTION
**Pull request by Frans06**
# Pull Request Template

## Description

Using numpy to optimize List operations

## Type of change

Please delete options that are not relevant.

- [X] New feature (non-breaking change which adds functionality)

## How Has This Been Tested?

- [X] make test

**Test Configuration**:
* OS: arch
* OS version: 5.8.14-arch1-1
* Python version: 3.8.6
* Architecture: x86

## Checklist:

- [X] My code follows the style guidelines of this project
- [X] I have performed a self-review of my own code
- [X] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [X] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [X] New and existing unit tests pass locally with my changes
- [X] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: pygamelib/engine.py [hunk 1]
```diff
@@ -91,7 +92,7 @@ def __init__(self, **kwargs):
         # The overlapped matrix is used as an invisible layer were overlapped
         # restorable items are parked momentarily (until the cell they were on
         # is free again).
-        self._overlapped_matrix = []
+        self._overlapped_matrix = np.array([])
```
Comment (arnauddupuis) [inline pygamelib/engine.py:95]: We should probably init self._matrix with an empty array too. I never paid attention to that until today...

**Review by arnauddupuis — CHANGES_REQUESTED** [review]
Hi,
Overall it looks great and lay the work for the next version of Board! Thanks.
A couple of points to change:

- You have to add the new numpy dependency in .legacy/requirements.txt (some tools I use do not support Pipfile yet).
-  why did you add a requirement.txt file at the root of the project? We use Pipfile and .legacy/requirements.txt for tools that do not support Pipfile yet.

Other than that: looks good to me!

File: pygamelib/engine.py (hunk 1 above)
Comment (Frans06) [reply pygamelib/engine.py:96]: Done

Comment (Frans06) [reply pygamelib/engine.py:96]: Making this better on the new version

**Comment by Frans06** [comment]
closes #118

File: .legacy/requirements.txt [hunk 2]
```diff
@@ -1,5 +1,8 @@
...
-readchar>=2.0.1
-blessed>=1.17.0
\ No newline at end of file
+-i https://pypi.org/simple
+appdirs==1.4.4
+blessed==1.17.11
+colorama==0.4.4
+numpy==1.19.2
+readchar==2.0.1
+six==1.15.0; python_version >= '2.7' and python_version not in '3.0, 3.1, 3.2, 3.3'
```
Comment (arnauddupuis) [inline .legacy/requirements.txt:7]: The python version are wrong: we are using f-strings extensively so we are only compatible with python 3.6+ as described in the setup.py file.

**Review by arnauddupuis — APPROVED** [review]
Aside from the version in .legacy/requirements.txt it looks good to me.

Thank you for your help!

**Pull request merged**